### PR TITLE
Kafka admin client health check rework

### PR DESCRIPTION
**Pull request by ozangunalp**
Incoming (consumer) channel topic-verification health checks:

Startup: Change listTopics to describeTopics to check the existence of the topic to subscribe. Breaking change
Readiness: Keep consumer assignment check.
Liveness: Keep

Outgoing (producer) channel topic-verification health checks:

Startup: Change listTopics to describeTopics to check the existence of the topic to publish. Breaking change
Readiness: Change listTopics with existence check to describeCluster to verify the cluster is still reachable. Breaking change
Liveness: Keep

Added `health-startup-topic-verification-disabled` and `health-readiness-topic-verification-disabled` flags to disable topic verification per probe per channel

### CODE REVIEW

File: documentation/src/main/docs/kafka/health.md [hunk 1]
```diff
@@ -1,54 +1,39 @@
...
-The Kafka connector reports the readiness and liveness of each channel
+The Kafka connector reports the startup, readiness and liveness of each channel
 managed by the connector.
 
 !!!note
     To disable health reporting, set the `health-enabled` attribute for the
     channel to `false`.
 
-## Readiness
+## Startup & Readiness
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:10]: While I love walls of text, I believe that a summary in a table would be good to have. WDYT?

File: documentation/src/main/docs/kafka/health.md [hunk 2]
```diff
@@ -1,54 +1,39 @@
...
 
-On the inbound side, two strategies are available to check the readiness
-of the application. The default strategy verifies that we have at least
-one active connection with the broker. This strategy is lightweight.
+By default, both inbound and outbound channels use underlying Kafka client metrics
+to check if at least one active connection exists with the broker.
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:13]:
```suggestion
to check if at least one active connection exists with a broker.
```

The current check verifies that one broker is reachable.

File: documentation/src/main/docs/kafka/health.md [hunk 3]
```diff
@@ -1,54 +1,39 @@
...
+This strategy is lightweight.
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:14]:
```suggestion
This strategy is lightweight and does not require remote interactions with the brokers.
```

File: documentation/src/main/docs/kafka/health.md [hunk 4]
```diff
@@ -1,54 +1,39 @@
...
-On the outbound side (writing records to Kafka), two strategies are also
-offered. The default strategy just verifies that the producer has at
-least one active connection with the broker.
-
-You can also enable another strategy by setting the
-`health-readiness-topic-verification` attribute to `true`. In this case,
-teh check verifies that
+`health-topic-verification-enabled` attribute to `true`.
+With this second strategy, the health checks use a Kafka Admin Client
+to access the broker. Retrieving the topics can be a lengthy
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:19]:
```suggestion
to access the broker and retrieve the list of existing topics. Retrieving this list can be a lengthy and expensive
```

File: documentation/src/main/docs/kafka/health.md [hunk 5]
```diff
@@ -1,54 +1,39 @@
...
+operation. You can configure a timeout using the
+`health-topic-verification-timeout` attribute. The default timeout is set to 2
+seconds.
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:22]:
```suggestion
seconds. Note that is the timeout is reached, the health check fails.
```

(Is that right?)

File: documentation/src/main/docs/kafka/health.md [hunk 6]
```diff
@@ -1,54 +1,39 @@
...
 
--   the broker is available
+!!!depreciation
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:24]: is that admonition working?

File: documentation/src/main/docs/kafka/health.md [hunk 7]
```diff
@@ -1,54 +1,39 @@
...
+## Startup & Readiness
 
-On the inbound side, two strategies are available to check the readiness
-of the application. The default strategy verifies that we have at least
-one active connection with the broker. This strategy is lightweight.
+By default, both inbound and outbound channels use underlying Kafka client metrics
+to check if at least one active connection exists with the broker.
+This strategy is lightweight.
 
 You can also enable another strategy by setting the
```
Comment (cescoffier) [inline documentation/src/main/docs/kafka/health.md:16]: I think it deserves its own section.

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaCheckpointCommit.java [hunk 8]
```diff
@@ -398,7 +398,7 @@ public LastStateStoredTooLongAgoException(TopicPartition topic, long time, long
         }
     }
 
-    static class CheckpointState<T> {
+    public static class CheckpointState<T> {
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaCheckpointCommit.java:401]: That seems to be important (while unrelated to that PR)

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaCheckpointCommit.java (hunk 8 above)
Comment (ozangunalp) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaCheckpointCommit.java:401]: This is why the revapi was complaining. But I didn't see that before

File: documentation/src/main/docs/kafka/health.md (hunk 6 above)
Comment (ozangunalp) [reply documentation/src/main/docs/kafka/health.md:24]: Replaced it with `!!!warning "Deprecated"`

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Need to be rebased

**Review by cescoffier — APPROVED** [review]

**Pull request merged**